src/installer.js

```diff
@@ -1,32 +1,100 @@
 const {exec} = require('@actions/exec')
 const path = require('path')
-const semver = require('semver')
+const core = require('@actions/core')
 
-module.exports = {installElixir, installOTP}
+/**
+ * Install Erlang/OTP.
+ *
+ * @param {string} osVersion
+ * @param {string} otpVersion
+ */
+async function installOTP(osVersion, otpVersion) {
+  return new Promise(async (resolve, reject) => {
+    const maybePlatformIsOKError = checkPlatform()
+    if (maybePlatformIsOKError !== 'ok') {
+      reject(new Error(maybePlatformIsOKError))
+      return
+    }
+    await exec(path.join(__dirname, 'install-otp'), [osVersion, otpVersion])
+      .then(() => {
+        resolve()
+      })
+      .catch((err) => {
+        core.warning(
+          `Installing Erlang/OTP ${otpVersion} over ${osVersion} failed. ` +
+            'It is possible the requested version does not exist',
+        )
+        reject(err)
+      })
+  })
+}
 
 /**
  * Install Elixir.
  *
- * @param {string} version
- * @param {string} otpMajor
+ * @param {string} elixirVersion
  */
-async function installElixir(version, otpMajor) {
-  if (process.platform === 'linux') {
-    const otpString = otpMajor ? `-otp-${otpMajor}` : ''
-    await exec(path.join(__dirname, 'install-elixir'), [version, otpString])
-  }
+async function installElixir(elixirVersion) {
+  return new Promise(async (resolve, reject) => {
+    const maybePlatformIsOKError = checkPlatform()
+    if (maybePlatformIsOKError !== 'ok') {
+      reject(new Error(maybePlatformIsOKError))
+      return
+    }
+    await exec(path.join(__dirname, 'install-elixir'), [elixirVersion])
+      .then(() => {
+        resolve()
+      })
+      .catch((err) => {
+        core.warning(
+          `Installing Elixir ${elixirVersion} failed. ` +
+            'It is possible the requested version does not exist',
+        )
+        reject(err)
+      })
+  })
 }
 
 /**
- * Install OTP.
+ * Install rebar3.
  *
- * @param {string} version
+ * @param {string} rebar3Version
  */
-async function installOTP(version, osVersion) {
-  if (process.platform === 'linux') {
-    await exec(path.join(__dirname, 'install-otp'), [version, osVersion])
-    return
+async function installRebar3(rebar3Version) {
+  return new Promise(async (resolve, reject) => {
+    const maybePlatformIsOKError = checkPlatform()
+    if (maybePlatformIsOKError !== 'ok') {
+      reject(new Error(maybePlatformIsOKError))
+      return
+    }
+    await exec(path.join(__dirname, 'install-rebar3'), [rebar3Version])
+      .then(() => {
+        resolve()
+      })
+      .catch((err) => {
+        core.warning(
+          `Installing rebar3 ${rebar3Version} failed. ` +
+            'It is possible the requested version does not exist',
+        )
+        reject(err)
+      })
+  })
+}
+
+function checkPlatform() {
+  if (
+    process.platform !== 'linux' &&
+    process.env.CI_TEST_ENV !== 'test' &&
+    process.platform !== 'darwin'
```

Comment:
This check should not be here, right?

Reply:
It should IMO. Locally I should be able to test on a Mac (especially since it works), but if you prefer I can take it out and only assume Linux based testing (will be kinda hindering for updating though - since I have to add it for testing, and remove it before pushing).

Reply:
I've reverted this change. It'll simply imply more force-pushes on my behalf to get stuff running properly. No biggie, though.

Reply:
How does it work on mac if erlang is compiled against linux?

Reply:
`setup-elixir.test.js` presents two types of tests:
* failing installs (e.g. inexisting file or input not accepted)
* version-based tests (testing against semver)
None of these require the builds to actually run, since we're interested in other conditions, so there's no difference in running them in Linux and Mac, except that without allow `process.platform` `darwin` you can't actually properly run these locally.

Reply:
In that case I would recommend unit testing that functionality directly so it doesn't go through this check.

Reply:
We have e.g. https://github.com/erlef/setup-elixir/pull/9/files#diff-3e35f8d82312f568d749abab56e430c703198c68e3b8468454005ac9bc806797R10. Do you prefer me to test 
```erlang
exec(path.join(__dirname, 'install-otp'), [
    osVersion,
    otpVersion,
  ])
```
directly? (as is, now, it's only affecting local testing, not remote)

Reply:
I'll move on this front, lest the answer is "yes" and then I lose the opportunity for a further review. 😄

Reply:
I've removed the validation (33c64db) since if we're not doing it for tests there's no need to do it any way (since `setup-elixir.js` already does it at start).